airflow/models/baseoperator.py

```diff
@@ -135,7 +135,10 @@ def _get_dag_defaults(dag: Optional["DAG"], task_group: Optional["TaskGroup"]) -
     dag_args = copy.copy(dag.default_args)
     dag_params = copy.deepcopy(dag.params)
     if task_group:
-        dag_args.update(task_group.default_args)
+        if not isinstance(task_group.default_args, Dict) and task_group.default_args is not None:
+            raise TypeError("default_args must be a dictionary")
+        else:
+            dag_args.update(task_group.default_args)
```

Comment:
```suggestion
        if task_group.default_args and not isinstance(task_group.default_args, collections.abc.Mapping):
            raise TypeError("default_args must be a mapping")
        dag_args.update(task_group.default_args)
```

Technically `default_args` does not need to be a dict, but only needs a [_dict-like interface_](https://docs.python.org/3/glossary.html#term-mapping).

Reply:
@uranusjr good to know thank you for reviewing this.